mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SpeechFacadeTest.kt

```diff
@@ -31,23 +50,48 @@ class SpeechFacadeTest {
   fun given_failingRecognizer_when_recognizes_then_returnsErrorInternal() = runTest {
     val facade =
         SpeechFacade(
-            FailingSpeechRecognizerFactory, FakeTextToSpeechFactory, emptyDataStoreFactory())
+            FailingSpeechRecognizerFactory,
+            FakeTextToSpeechFactory,
+            FakeSoundPlayer,
+            emptyDataStoreFactory())
     assertThat(facade.recognize()).isEqualTo(Failure.Internal)
   }
 
   @Test
   fun given_successfulRecognizer_when_recognizes_then_returnsResults() = runTest {
     val facade =
         SpeechFacade(
-            UnknownCommandSpeechRecognizerFactory, FakeTextToSpeechFactory, emptyDataStoreFactory())
+            UnknownCommandSpeechRecognizerFactory,
+            FakeTextToSpeechFactory,
+            FakeSoundPlayer,
+            emptyDataStoreFactory())
     assertThat(facade.recognize()).isEqualTo(Success(SuccessfulSpeechRecognizer.Results))
   }
 
   @Test
   fun given_fakeTextToSpeech_when_synthesizing_then_nothing_happens() = runTest {
     val facade =
         SpeechFacade(
-            UnknownCommandSpeechRecognizerFactory, FakeTextToSpeechFactory, emptyDataStoreFactory())
+            UnknownCommandSpeechRecognizerFactory,
+            FakeTextToSpeechFactory,
+            FakeSoundPlayer,
+            emptyDataStoreFactory())
     assertThat(facade.synthesize("Pawnies")).isEqualTo(Unit)
   }
+
+  @Test
+  fun given_speechFacade_when_synthesizing_then_playSoundCalled() = runTest {
+    var called = false
+    val mediaPlayer = FakeMediaPlayer(startCallback = { called = true })
+    val context = mockk<Context>()
+    val soundPlayer = AndroidSoundPlayer(context = context, mediaPlayer = mediaPlayer)
+    val facade =
+        SpeechFacade(
+            UnknownCommandSpeechRecognizerFactory,
+            FakeTextToSpeechFactory,
+            soundPlayer,
+            emptyDataStoreFactory())
+    facade.synthesize("Pawnies")
+    assertThat(called).isTrue()
+  }
```

Comment:
This should not be in `SpeechFacadeTest`. It looks like you're testing both the `AndroidSoundPlayer` and the `SpeechFacade` simultaneously.